keys.go

```diff
@@ -260,6 +272,41 @@ func (c *Client) SetKeyRing(ctx context.Context, idOrAlias, newKeyRingID string)
 	return &response.Keys[0], nil
 }
 
+// CreateKeyWithPolicyOverride creates a new KP key with given key policy details
+func (c *Client) CreateKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, extractable bool, aliases []string, policy Policy) (*Key, error) {
+	return c.CreateImportedKeyWithPolicyOverride(ctx, name, expiration, "", "", "", extractable, aliases, policy)
+}
+
+// CreateImportedKeyWithPolicyOverride creates a new Imported KP key from the given key material and with given key policy details
+func (c *Client) CreateImportedKeyWithPolicyOverride(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string, extractable bool, aliases []string, policy Policy) (*Key, error) {
+	key := c.createKeyTemplate(ctx, name, expiration, payload, encryptedNonce, iv, extractable, aliases, AlgorithmRSAOAEP256)
+
+	key.Rotation = policy.Rotation
```

Comment:
It wont, I have tested it.. I am just setting `key.Rotation=nil` in case `policy.Rotation` is nil. Also `policy' itself would never be nil as we are passing the structure itself.

Reply:
Anyways, I have added this function inside createKeyTemplate and added this check as well.